pkg/operator/hive/hive.go

```diff
@@ -211,3 +215,23 @@ func (r *ReconcileHiveConfig) includeAdditionalCAs(hLog log.FieldLogger, h *reso
 
 	return nil
 }
+
+func (r *ReconcileHiveConfig) includeGlobalPullSecret(hLog log.FieldLogger, h *resource.Helper, instance *hivev1.HiveConfig, hiveDeployment *appsv1.Deployment) error {
+	if instance.Spec.GlobalPullSecret == nil {
+		hLog.Debug("GlobalPullSecret is not provided in HiveConfig, it will not be deployed")
+		return nil
+	}
+	globalPullSecretEnvVars := []corev1.EnvVar{
+		{
+			Name: "GLOBAL_PULL_SECRET",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: instance.Spec.GlobalPullSecret,
```

Comment:
@dgoodwin looks this usage of `instance.Spec.GlobalPullSecret` is wrong. I am getting below error while compiling 

`pkg/operator/hive/hive.go:229:6: cannot use instance.Spec.GlobalPullSecret (type *"github.com/openshift/hive/vendor/k8s.io/api/core/v1".LocalObjectReference) as type "github.com/openshift/hive/vendor/k8s.io/api/core/v1".LocalObjectReference in field value`